services/pkg/catalogue/opa.go

```diff
@@ -0,0 +1,79 @@
+// Copyright (c) 2024 - for information on the respective copyright owner
+// see the NOTICE file and/or the repository https://github.com/carbynestack/thymus.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+package catalogue
+
+import (
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+)
+
+const BasePath = "v1/policies"
+
+// Result is the response from the OPA service
+type Result struct {
+	Policies []Policy `json:"result"`
+}
+
+// Policy represents a policy in the OPA service
+type Policy struct {
+	ID   string `json:"id"`
+	Code string `json:"raw"`
+	Ast  string `json:"-"` // Will be ignored
+}
+
+// fetchPoliciesFromOPA fetches all policies from the OPA service. Note that
+// there is unfortunately no pagination support.
+func fetchPoliciesFromOPA(opaSvcUrl string) (policies map[string]Policy, error error) {
+	resp, err := http.Get(fmt.Sprintf("%s/%s", opaSvcUrl, BasePath))
+	if err != nil {
+		return nil, err
+	}
+	defer closeBody(resp.Body, &error)
+
+	body, err := io.ReadAll(resp.Body)
+	if err != nil {
+		return nil, err
+	}
+
+	result := Result{}
+	err = json.Unmarshal(body, &result)
+	if err != nil {
+		return nil, err
+	}
+
+	policies = make(map[string]Policy)
+	for _, policy := range result.Policies {
+		policies[policy.ID] = policy
+	}
+	return policies, nil
+}
+
+// fetchPolicyFromOPA fetches a single policy from the OPA service by its ID.
+// Note that there is no way to fetch a single policy by ID, so we fetch all.
```

Comment:
but one could simply call `v1/policies/<POLICY_ID>`

Reply:
see [get-a-policy](https://www.openpolicyagent.org/docs/latest/rest-api/#get-a-policy)

Reply:
Don't know how I have missed that. Used now.